VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/select-hearing-layout/select-hearing-layout.component.ts

```diff
@@ -39,8 +49,8 @@ export class SelectHearingLayoutComponent implements OnInit {
         const text = this.accordionOpenAllElement.innerHTML;
         if (!text.startsWith('<')) {
             const originalText = text.split('<')[0];
-            const currentText = originalText.toLowerCase().split(' ').join('-').trim();
-            const translated = `<span>${this.translateService.instant(`select-hearing-layout.${currentText}`)}</span>`;
+            this.currentText = originalText.toLowerCase().split(' ').join('-').trim();
+            const translated = `<span>${this.translateService.instant(`select-hearing-layout.${this.currentText}`)}</span>`;
```

Comment:
We can test that currentText is set correctly now. Maybe make it more clear what this does, this is because we have no control over that button right? So it will always be either open-all or close-all.

Reply:
Could some of this be another function since we do a very similar process in both this function and the subscribe?

Reply:
yea I can add assertion to check what the expected current text is in all the relevant tests, shall I do that?

Reply:
regarding re using that if block, the logic appears to be specific to the lifecycle of the component itself. Initially the innerhtml doesn't appear to start with a "<" but in my case the accordion will have that and also, even if I were to use the the same logic it appears after using "split" on the innerHTML, the part of text that I require to translate no longer is in the the array index of 0. So instead I used innerText in ngoninit. Might be easier to show it rather than explaining here